sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -370,75 +419,96 @@ public static <T> Flux<? extends IDocumentQueryExecutionContext<T>> createSpecia
             return Flux.error(validationError);
         }
 
-        boolean getLazyFeedResponse = queryInfo.hasTop();
+        boolean getLazyFeedResponse = Boolean.FALSE;
 
-        // We need to compute the optimal initial age size for non-streaming order-by queries
-        if (queryInfo.hasNonStreamingOrderBy()) {
-            // Validate the TOP or LIMIT for non-streaming order-by queries
-            if (!queryInfo.hasTop() && !queryInfo.hasLimit() && queryInfo.getTop() < 0 && queryInfo.getLimit() < 0) {
-                throw new NonStreamingOrderByBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
-                    "Executing a vector search query without TOP or LIMIT can consume a large number of RUs" +
-                        "very fast and have long runtimes. Please ensure you are using one of the above two filters" +
-                        "with you vector search query.");
-            }
-            // Validate the size of TOP or LIMIT against MaxItemSizeForVectorSearch
-            int maxLimit = Math.max(queryInfo.hasTop() ? queryInfo.getTop() : 0,
-                queryInfo.hasLimit() ? queryInfo.getLimit() : 0);
-            int maxItemSizeForVectorSearch = Math.max(Configs.getMaxItemCountForVectorSearch(),
-                qryOptAccessor.getMaxItemCountForVectorSearch(cosmosQueryRequestOptions));
-            if (maxLimit > maxItemSizeForVectorSearch) {
-                throw new NonStreamingOrderByBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
-                    "Executing a vector search query with TOP or LIMIT larger than the maxItemSizeForVectorSearch " +
-                        "is not allowed");
+        if (hybridSearchQueryInfo!=null) {
+            // Validate the TOP for non-streaming order-by queries
+            if (!hybridSearchQueryInfo.hasTake()) {
+                throw new HybridSearchBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
+                    "Executing a hybrid or full text query without Top can consume a large number of RUs " +
+                        "very fast and have long runtimes. Please ensure you are using the above filter " +
+                        "with your hybrid or full text search query.");
             }
-            // Set initialPageSize based on the smallest of TOP or LIMIT
-            if (queryInfo.hasTop() || queryInfo.hasLimit()) {
-                int pageSizeWithTopOrLimit = Math.min(queryInfo.hasTop() ? queryInfo.getTop() : Integer.MAX_VALUE,
-                                           queryInfo.hasLimit() && queryInfo.hasOffset() ?
-                                               queryInfo.getLimit() + queryInfo.getOffset() : Integer.MAX_VALUE);
-                initialPageSize = pageSizeWithTopOrLimit;
+
+            if (hybridSearchQueryInfo.hasSkip() && !hybridSearchQueryInfo.hasTake()) {
+                throw new HybridSearchBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
```

Comment:
curious: I thought backend will make sure OFFSET and LIMIT both there for offset limit query, does this check need to be done from SDK side?

Reply:
I am doing a check if offset and limit are there, then limit - offset should not be less than or equal to zero. So, this check here makes sense.

Reply:
That is a completely weird constrtaint thought. It makes it impossible to use skip/limit for pagination - the single purpose why it exists? Is this a temporary limitation or supposed to stay there long-term?